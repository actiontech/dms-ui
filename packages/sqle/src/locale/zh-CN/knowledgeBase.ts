// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: '知识库',
  searchResults: {
    backToKnowledgeBase: '返回知识库',
    searchPlaceholder: '请输入关键字搜索你感兴趣的SQL语法'
  },
  search: {
    retrieveKnowledgeGraph: '检索知识图谱',
    selectedTags: '已选择标签'
  }
};
