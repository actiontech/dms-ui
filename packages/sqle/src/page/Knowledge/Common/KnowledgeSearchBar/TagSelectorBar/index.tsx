import { useTranslation } from 'react-i18next';
import { DownOutlined } from '@actiontech/icons';
import knowledge_base from '@actiontech/shared/lib/api/sqle/service/knowledge_base';
import { useRequest, useControllableValue } from 'ahooks';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { Popover, Checkbox, Empty } from 'antd';
import { BasicInput, BasicTag, EmptyBox } from '@actiontech/shared';
import {
  KnowledgeSearchTagsPopoverStyleWrapper,
  KnowledgeTagSelectorBarStyleWrapper
} from '../style';
import { useMemo, useState } from 'react';
import { TagSelectorBarProps } from '../index.type';

const TagSelectorBar: React.FC<TagSelectorBarProps> = (props) => {
  const { t } = useTranslation();

  const [selectedTags, setSelectedTags] = useControllableValue<
    TagSelectorBarProps['value']
  >(props, {
    defaultValue: []
  });

  const [tagSearchText, setTagSearchText] = useState('');

  const { data: tagOptions } = useRequest(() =>
    knowledge_base.getKnowledgeBaseTagList().then((res) => {
      if (res.data.code === ResponseCode.SUCCESS) {
        return (
          res.data.data?.map((tag) => ({
            label: tag.name,
            value: tag.name
          })) ?? []
        );
      }
      return [];
    })
  );

  const handleTagRemove = (tagToRemove: string) => {
    const filteredTags = selectedTags?.filter((tag) => tag !== tagToRemove);
    setSelectedTags?.(filteredTags);
  };

  const filteredTagOptions = useMemo(() => {
    return (
      tagOptions?.filter((option) =>
        option.label?.toLowerCase().includes(tagSearchText.toLowerCase())
      ) ?? []
    );
  }, [tagSearchText, tagOptions]);

  return (
    <KnowledgeTagSelectorBarStyleWrapper size={4} wrap>
      <Popover
        trigger="click"
        placement="bottomLeft"
        arrow={false}
        overlayInnerStyle={{
          padding: 0
        }}
        content={
          <KnowledgeSearchTagsPopoverStyleWrapper direction="vertical">
            <BasicInput
              size="small"
              value={tagSearchText}
              placeholder={t('common.search')}
              onChange={(e) => setTagSearchText(e.target.value)}
              allowClear
            />
            <EmptyBox
              if={!!filteredTagOptions?.length}
              defaultNode={<Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />}
            >
              <Checkbox.Group
                value={selectedTags}
                onChange={(tags) => {
                  setSelectedTags?.(tags as string[]);
                }}
              >
                {filteredTagOptions.map((option) => (
                  <Checkbox key={option.value} value={option.value}>
                    <span title={option.label}>{option.label}</span>
                  </Checkbox>
                ))}
              </Checkbox.Group>
            </EmptyBox>
          </KnowledgeSearchTagsPopoverStyleWrapper>
        }
      >
        <div className="popover-target">
          <span>
            {t('knowledgeBase.search.selectedTags')}({selectedTags?.length})
          </span>
          <DownOutlined width={20} height={20} />
        </div>
      </Popover>
      {selectedTags?.map((tag) => (
        <BasicTag
          key={tag}
          closable
          size="small"
          onClose={() => handleTagRemove(tag)}
        >
          {tag}
        </BasicTag>
      ))}
    </KnowledgeTagSelectorBarStyleWrapper>
  );
};

export default TagSelectorBar;
