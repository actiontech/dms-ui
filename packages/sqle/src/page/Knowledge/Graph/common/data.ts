import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
