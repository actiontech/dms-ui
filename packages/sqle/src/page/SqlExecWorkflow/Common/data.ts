export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';
