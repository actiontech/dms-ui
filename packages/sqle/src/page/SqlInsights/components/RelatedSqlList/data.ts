import { GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum } from '@actiontech/shared/lib/api/sqle/service/SqlInsight/index.enum';
import { t } from '../../../../locale';

export const RelatedSqlFilterSourceOptions = [
  {
    label: t('sqlInsights.relatedSqlList.source.order'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workflow
  },
  {
    label: t('sqlInsights.relatedSqlList.source.sqlManage'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.sql_manage
  },
  {
    label: t('sqlInsights.relatedSqlList.source.workbench'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workbench
  }
];
