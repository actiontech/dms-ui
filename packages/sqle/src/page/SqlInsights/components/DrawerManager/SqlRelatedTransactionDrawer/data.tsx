import { TransactionInfoLockTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { t } from '../../../../../locale';

export const TransactionInfoLockTypeEnumDict = {
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.exclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.shared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionShared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.sharedIntentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.ROW_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.rowLock'
  ),
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.tableLock'
  ),
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.metadataLock'
  )
};

export const TransactionInfoLockTypeEnumFlagDict = {
  [TransactionInfoLockTypeEnum.SHARED]: 'S',
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: 'IS',
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: 'IX',
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: 'SIX',
  [TransactionInfoLockTypeEnum.ROW_LOCK]: 'RL',
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: 'TL',
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: 'ML',
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: 'X'
};
