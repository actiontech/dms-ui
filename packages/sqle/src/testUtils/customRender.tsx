import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
