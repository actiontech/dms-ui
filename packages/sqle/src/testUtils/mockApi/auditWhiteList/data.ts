import { IAuditWhitelistResV1 } from '@actiontech/shared/lib/api/sqle/service/common';
import { CreateAuditWhitelistReqV1MatchTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export const auditWhiteListMockData: IAuditWhitelistResV1[] = [
  {
    audit_whitelist_id: 1,
    desc: 'test1',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.exact_match,
    value: 'SELECT 1;',
    matched_count: 1,
    last_match_time: '2024-08-22T11:04:29.543+08:00'
  },
  {
    audit_whitelist_id: 2,
    desc: 'test2',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.fp_match,
    value: 'SELECT 1;',
    matched_count: 10,
    last_match_time: undefined
  },
  {
    audit_whitelist_id: 3,
    desc: 'test3',
    match_type: CreateAuditWhitelistReqV1MatchTypeEnum.exact_match,
    value: 'SELECT 1;',
    matched_count: 0
  },
  {
    audit_whitelist_id: 4,
    desc: 'test4'
  }
];
