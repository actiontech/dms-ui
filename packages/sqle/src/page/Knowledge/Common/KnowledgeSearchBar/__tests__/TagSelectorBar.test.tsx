import { act, cleanup, screen, fireEvent } from '@testing-library/react';
import TagSelectorBar from '../TagSelectorBar';
import { superRender } from '../../../../../testUtils/customRender';
import knowledgeBase from '../../../../../testUtils/mockApi/knowledgeBase';
import { mockKnowledgeBaseTagListData } from '../../../../../testUtils/mockApi/knowledgeBase/data';
import {
  getAllBySelector,
  getBySelector
} from '@actiontech/shared/lib/testUtil/customQuery';

describe('TagSelectorBar', () => {
  let getKnowledgeBaseTagList: jest.SpyInstance;

  beforeEach(() => {
    jest.useFakeTimers();
    getKnowledgeBaseTagList = knowledgeBase.getKnowledgeBaseTagList();
  });

  afterEach(() => {
    jest.useRealTimers();
    cleanup();
  });

  it('render init snap', () => {
    const { container } = superRender(
      <TagSelectorBar value={['RAND']} onChange={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
    expect(getKnowledgeBaseTagList).toHaveBeenCalled();
  });

  it('render select tag', async () => {
    const onChangeSpy = jest.fn();
    const { baseElement } = superRender(
      <TagSelectorBar value={[]} onChange={onChangeSpy} />
    );
    expect(getKnowledgeBaseTagList).toHaveBeenCalledTimes(1);
    await act(async () => jest.advanceTimersByTime(3000));
    fireEvent.click(screen.getByText('已选择标签(0)'));
    await act(async () => jest.advanceTimersByTime(0));
    expect(baseElement).toMatchSnapshot();

    fireEvent.change(getBySelector('.ant-input'), {
      target: {
        value: 'rand'
      }
    });
    await act(async () => jest.advanceTimersByTime(0));
    expect(screen.getByText('RAND')).toBeInTheDocument();
    expect(getAllBySelector('.hidden-checkbox')).toHaveLength(
      mockKnowledgeBaseTagListData.length - 1
    );
    fireEvent.click(getAllBySelector('.ant-checkbox-input')[0]);
    await act(async () => jest.advanceTimersByTime(0));
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
    expect(onChangeSpy).toHaveBeenCalledWith(['RAND']);
  });

  it('render delete tag', async () => {
    const onChangeSpy = jest.fn();
    superRender(<TagSelectorBar value={['RAND']} onChange={onChangeSpy} />);
    expect(getKnowledgeBaseTagList).toHaveBeenCalledTimes(1);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(screen.getByText('已选择标签(1)')).toBeInTheDocument();
    expect(screen.getByText('RAND')).toBeInTheDocument();

    fireEvent.click(getBySelector('.anticon-close'));
    await act(async () => jest.advanceTimersByTime(0));
    expect(onChangeSpy).toHaveBeenCalledTimes(1);
    expect(onChangeSpy).toHaveBeenCalledWith([]);
  });
});
