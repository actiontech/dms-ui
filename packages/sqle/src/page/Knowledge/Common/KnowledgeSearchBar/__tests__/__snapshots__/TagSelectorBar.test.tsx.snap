// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSelectorBar render init snap 1`] = `
<div>
  <div
    class="ant-space ant-space-horizontal ant-space-align-center css-5rjiyz"
    style="flex-wrap: wrap; margin-bottom: -4px;"
  >
    <div
      class="ant-space-item"
      style="margin-right: 4px; padding-bottom: 4px;"
    >
      <div
        class="popover-target"
      >
        <span>
          已选择标签
          (
          1
          )
        </span>
        <svg
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8 8.781 3.3-3.3.943.943L8 10.667 3.757 6.424l.943-.943z"
            fill="#C3C6CD"
          />
        </svg>
      </div>
    </div>
    <div
      class="ant-space-item"
      style="padding-bottom: 4px;"
    >
      <span
        class="ant-tag ant-tag-default basic-tag-wrapper basic-small-tag-wrapper css-d31ub5"
      >
        RAND
        <span
          aria-label="close"
          class="anticon anticon-close ant-tag-close-icon"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="close"
            fill="currentColor"
            fill-rule="evenodd"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`TagSelectorBar render select tag 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center css-5rjiyz"
      style="flex-wrap: wrap; margin-bottom: -4px;"
    >
      <div
        class="ant-space-item"
        style="padding-bottom: 4px;"
      >
        <div
          class="popover-target ant-popover-open"
        >
          <span>
            已选择标签
            (
            0
            )
          </span>
          <svg
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m8 8.781 3.3-3.3.943.943L8 10.667 3.757 6.424l.943-.943z"
              fill="#C3C6CD"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-popover ant-zoom-big-appear ant-zoom-big-appear-prepare ant-zoom-big ant-popover-placement-bottomLeft"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box;"
    >
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-inner"
          role="tooltip"
          style="padding: 0px;"
        >
          <div
            class="ant-popover-inner-content"
          >
            <div
              class="ant-space ant-space-vertical css-yta4rc"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 8px;"
              >
                <span
                  class="ant-input-affix-wrapper basic-input-wrapper css-1pd1cd5 ant-input-affix-wrapper-sm"
                >
                  <input
                    class="ant-input ant-input-sm"
                    placeholder="搜索"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-clear-icon ant-input-clear-icon-hidden"
                      role="button"
                      tabindex="-1"
                    >
                      <svg
                        fill="currentColor"
                        height="14"
                        viewBox="0 0 14 14"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m7 6.175 2.887-2.887.825.825L7.825 7l2.887 2.888-.825.824L7 7.825l-2.888 2.887-.824-.824L6.175 7 3.288 4.113l.824-.825z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
              <div
                class="ant-space-item"
              >
                <div
                  class="ant-checkbox-group"
                >
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="RAND"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="RAND"
                      >
                        RAND
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="YEAR"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="YEAR"
                      >
                        YEAR
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="HINT"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="HINT"
                      >
                        HINT
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="ALTER_TABLE"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="ALTER_TABLE"
                      >
                        ALTER_TABLE
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="EXPLAIN_FORMAT_JSON"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="EXPLAIN_FORMAT_JSON"
                      >
                        EXPLAIN_FORMAT_JSON
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="SUBQUERY"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="SUBQUERY"
                      >
                        SUBQUERY
                      </span>
                    </span>
                  </label>
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        type="checkbox"
                        value="FOREIGN_KEY"
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      <span
                        title="FOREIGN_KEY"
                      >
                        FOREIGN_KEY
                      </span>
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
